src/Arcus.Testing.Tests.Integration/Storage/Fixture/MongoDbTestContext.cs

```diff
@@ -101,18 +101,48 @@ public string WhenCollectionNameUnavailable()
         }
 
         /// <summary>
-        /// Provides a collection name that does exists in the MongoDb database.
+        /// Provides a collection name that does exist in the MongoDb database.
         /// </summary>
         public async Task<string> WhenCollectionNameAvailableAsync()
         {
             string collectionName = WhenCollectionNameUnavailable();
             _logger.LogTrace("[Test] create MongoDb collection '{CollectionName}' outside the fixture's scope", collectionName);
 
-            await _database.CreateCollectionAsync(collectionName);
+            await WhenMongoDbAvailableAsync(
+                () => _database.CreateCollectionAsync(collectionName),
+                $"[Test] cannot create MongoDb collection '{collectionName}' outside the fixture's scope, due to a high-rate failure");
 
             return collectionName;
         }
 
+        /// <summary>
+        /// Runs an <paramref name="mongoDbOperationAsync"/> within a MongoDb connection, with transient-failure retry.
+        /// </summary>
+        /// <param name="mongoDbOperationAsync">The operation to run against a MongoDb database.</param>
+        /// <param name="errorMessage">The custom user message that describes the <paramref name="mongoDbOperationAsync"/>.</param>
+        internal static async Task WhenMongoDbAvailableAsync(Func<Task> mongoDbOperationAsync, string errorMessage)
+        {
+            await WhenMongoDbAvailableAsync(async () =>
+            {
+                await mongoDbOperationAsync();
+                return 0;
+
+            }, errorMessage);
+        }
+
+        /// <summary>
+        /// Runs an <paramref name="mongoDbOperationAsync"/> within a MongoDb connection, with transient-failure retry.
+        /// </summary>
+        /// <param name="mongoDbOperationAsync">The operation to run against a MongoDb database.</param>
+        /// <param name="errorMessage">The custom user message that describes the <paramref name="mongoDbOperationAsync"/>.</param>
+        internal static async Task<TResult> WhenMongoDbAvailableAsync<TResult>(Func<Task<TResult>> mongoDbOperationAsync, string errorMessage)
+        {
+            return await Poll.Target<TResult, MongoCommandException>(mongoDbOperationAsync)
+                             .When(ex => ex.Message.Contains("high rate") || ex.ErrorMessage.Contains("high rate"))
```

Comment:
New feature also allows us to do this: which was needed to make the MongoDb integration test more stable in parallel contexts (like multiple PR's).